### PR TITLE
Bug 1862426: gather the audit logs for oauth apiserver

### PR DESCRIPTION
**Pull request by sanchezl**
* simplified logic
* added progress logging
* also capture oauth-apiserver logs
* add `SINCE_TIMEFRAME` env, defaulted to `1w`

### CODE REVIEW

**Comment by sferich888** [comment]
@sanchezl we need to review this in the context of https://github.com/openshift/must-gather/pull/143 (and what it is trying to achieve). https://github.com/openshift/must-gather/pull/164 is also going to be affected by this.

File: collection-scripts/gather_audit_logs
Comment (sferich888) [inline collection-scripts/gather_audit_logs:10]: Can you add `SINCE_TIMEFRAME=${5:-1w}` to this?

File: collection-scripts/gather_audit_logs
Comment (sferich888) [inline collection-scripts/gather_audit_logs:21]: Can you add  `--since ${SINCE_TIMEFRAME}` to this?

File: collection-scripts/gather_audit_logs
Comment (p0lyn0mial) [inline collection-scripts/gather_audit_logs:11]: @sanchezl please add `oauth-apiserver` to the list and close https://github.com/openshift/must-gather/pull/164

File: collection-scripts/gather_audit_logs
Comment (sanchezl) [reply collection-scripts/gather_audit_logs]: @sferich888 , I have added the `--since` parameter, but it doesn't seem to actually do anything.

File: collection-scripts/gather_audit_logs
Comment (sanchezl) [reply collection-scripts/gather_audit_logs]: Done

**Comment by sferich888** [comment]
@sanchezl we need to make sure we look into https://bugzilla.redhat.com/show_bug.cgi?id=1861746 
Closing in favor of https://github.com/openshift/must-gather/pull/144

**Comment by sferich888** [comment]
@sanchezl also note RFE-309 is something that will be closed by this.

File: collection-scripts/gather_audit_logs
Comment (sanchezl) [reply collection-scripts/gather_audit_logs]: The `--since` parameter is only applied to journal logs. It it ignored for `--path` logs, which are simply served by an instance of `http.FileServer`.

Comment (sferich888) [reply collection-scripts/gather_audit_logs:22]: What other ways can we limit the collection? 

Is an awk script needed? 

`#!/usr/bin/gawk -f BEGIN { starttime = mktime(starttime) endtime = mktime(endtime) } func in_range(n, start, end) { return start <= n && n < end } match($0, /^([0-9]{4})-([0-9]{2})-([0-9]{2})\s/, m) && in_range(mktime(m[1] " " m[2] " " m[3] " 00 00 00"), `

Source: https://askubuntu.com/questions/800769/how-to-get-text-from-range-of-dates-using-grep-sed-in-large-text-file

Comment (sanchezl) [reply collection-scripts/gather_audit_logs:23]: @sferich888 I think I can work something out, but as a followup.

**Comment by sanchezl** [comment]
Waiting for https://github.com/openshift/origin/pull/25369.

File: collection-scripts/gather_audit_logs
Comment (dougsland) [reply collection-scripts/gather_audit_logs:23]: In others scripts I see the full path of command, like:/usr/bin/oc

**Review by soltysh — APPROVED** [review]
/lgtm

File: collection-scripts/gather
Comment (deads2k) [inline collection-scripts/gather:45]: you can't do this until the e2e test works.

Comment (soltysh) [reply collection-scripts/gather:45]: that e2e you're talking about already explicitly calls this script see https://github.com/openshift/origin/blob/1a83021f4dbc91de606a71d3a27cd80664aae26c/test/extended/cli/mustgather.go#L160 so it shouldn't block this PR

**Comment by deads2k** [comment]
wait for https://github.com/openshift/origin/pull/25493 to merge

**Comment by deads2k** [comment]
/test all
/lgm
/hold

need to see green tests including audit (not skipped) before merge.

**Comment by sanchezl** [comment]
/retest 

Now that https://github.com/openshift/origin/pull/25493 has merged

**Comment by sanchezl** [comment]
/test all

Now that https://github.com/openshift/origin/pull/25493 has merged.

**Review by soltysh — APPROVED** [review]
Green audit e2e
/lgtm

**Pull request merged**
